jdatetime/__init__.py

```diff
@@ -645,6 +646,38 @@ def aslocale(self, locale):
 date.max = date(MAXYEAR, 12, 30)
 
 
+if sys.version_info[:2] < (3, 7):
+    # Prior to Python 3.7 all characters are escaped by re.escape
+    _SPECIAL_CHARS_MAP = {
+        ord(i): '\\' + str(i) for i in '()[]{}?*+-|^$\\.&~# \t\n\r\v\f'}
```

Comment:
```suggestion
    _SPECIAL_CHARS_MAP = {
        ord(i): '\\' + str(i) for i in '()[]{}?*+-|^$\\.&~# \t\n\r\v\f'
    }
```
Please fix it in other places as well.

Reply:
Done. (This part of code is removed from next patch. )